crates/web5/src/apid/jwk.rs

```diff
@@ -0,0 +1,50 @@
+use base64::{engine::general_purpose, Engine};
+use serde::{Deserialize, Serialize};
+use sha2::{Digest, Sha256};
+
+#[derive(Serialize, Deserialize, Default, Clone, Debug, PartialEq)]
+pub struct Jwk {
+    pub alg: String,
+    pub kty: String,
+    pub crv: String,
+    pub d: Option<String>,
+    pub x: String,
+    pub y: Option<String>,
+}
+
+#[derive(thiserror::Error, Debug, Clone, PartialEq)]
+pub enum JwkError {
+    #[error("thumbprint computation failed {0}")]
+    ThumbprintFailed(String),
+}
+
+type Result<T> = std::result::Result<T, JwkError>;
+
+impl Jwk {
+    pub fn compute_thumbprint(&self) -> Result<String> {
```

Comment:
`compute_thumbprint()` isn't in the APID, but it's here as an implementation detail

over time we should consider adding it to the APID (but keeping lean for now)